weld/type_inference.rs

```diff
@@ -338,6 +338,10 @@ fn infer_locally(expr: &mut PartialExpr, env: &mut TypeMap) -> WeldResult<bool>
                 }
                 changed |= try!(push_type(&mut expr.ty, &data.ty, "Sort"));
                 Ok(changed)
+            } else if data.ty == Unknown {
+                // TODO : The type inferencing is giving unkowns for
```

Comment:
This is a hack for now. 
For some reason, it's not inferring the type of a let assigned to a vector of structs.

Reply:
A @sppalkia 
Should I get rid of this line as well now with the type inference changes?